pkg/kubectl/cmd/set/set_env.go

```diff
@@ -278,7 +293,15 @@ func (o *EnvOptions) RunEnv(f cmdutil.Factory) error {
 							},
 						},
 					}
-					env = append(env, envVar)
+					if len(o.Keys) > 0 {
```

Comment:
Can you make a helper out of it, instead of copy&pasting it for secret here and config map below.